cmd/snap-update-ns/update.go

```diff
@@ -77,21 +75,36 @@ func applySystemFstab(ctx MountProfileUpdateContext) error {
 }
 
 func applyUserFstab(ctx MountProfileUpdateContext) error {
+	unlock, err := ctx.Lock()
+	if err != nil {
+		return err
+	}
+	defer unlock()
+
 	desired, err := ctx.LoadDesiredProfile()
 	if err != nil {
 		return err
 	}
 	debugShowProfile(desired, "desired mount profile")
 
-	current, err := ctx.LoadCurrentProfile()
+	currentBefore, err := ctx.LoadCurrentProfile()
 	if err != nil {
 		return err
 	}
-	debugShowProfile(current, "current mount profile")
+	debugShowProfile(currentBefore, "current mount profile (before applying changes)")
```

Comment:
Should this say "current user mount profile..."?

Reply:
No, because the whole point is to unify this into one function.